src/Symfony/Component/Security/Core/Authorization/Voter/Context/VoterContextFactory.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Core\Authorization\Voter\Context;
+
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Authorization\Voter\Report\VoteReportCollectorInterface;
+use Symfony\Component\Security\Core\Authorization\Voter\VoterInterface;
+use Symfony\Component\Translation\TranslatorInterface;
+
+/**
+ * VoterContextFactory creates new {@link VoterContext} instances.
+ *
+ * @author Maxime Perrimond <max.perrimond@gmail.com>
+ */
+class VoterContextFactory implements VoterContextFactoryInterface
+{
+    /**
+     * @var VoteReportCollectorInterface
+     */
+    private $collector;
+
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @var string|null
+     */
+    private $translationDomain;
+
+    /**
+     * Constructor.
+     *
+     * @param VoteReportCollectorInterface $collector
+     * @param TranslatorInterface          $translator        The translator
+     * @param string|null                  $translationDomain The translation domain to
+     *                                                        use for translating
+     *                                                        report messages
+     */
+    public function __construct(VoteReportCollectorInterface $collector, TranslatorInterface $translator, $translationDomain = null)
```

Comment:
`Constructor.` can be removed, it adds nothing of value. The `$translationDomain` should probably just go on 1 line.